apps/daymoon/app.js

```diff
@@ -0,0 +1,214 @@
+    let k = mp.fraction;
+    // k is the percent along the equator of the terminator
+    const pts = Math.min(pos.r >> 1, 32);
+    // this gives r/2 pts on the way down and up, capped at 64 total for polyfill
+    let a = [],
+        b = [],
+        s1 = 1,
+        s2 = 0;
+    // scale s1 is 1 or -1 for fixed edge of the shadow; defined via case switches below
+    // scale s2 factor for the moving edge of the shadow
+    // need to do some computation to simplify for new/full moon if k 'close enough' to 0 or 1/-1
+    //  
+    let isWaxing = (mp.phase < 0.5);
+    s1 = isWaxing ? -1 : 1;
+    s2 = isWaxing ? 1 - 2 * k : 2 * k - 1;
+    let tr = (pos.r + 0.5);
+    for (var i = 0; i < pts; i++) {
+        // down stroke on the outer shadow
+        var t = i * Math.PI / (pts + 1); //pts+1 so we leave the last point for the starting of going up
+        let cirX = Math.sin(t) * tr;
+        let cirY = Math.cos(t) * tr;
+        a.push(pos.x + s1 * cirX); //x
+        a.push(pos.y + cirY); //y
+        b.push(pos.x + s2 * cirX); //x for shadow edge
+        b.push(pos.y - cirY); //y going up for shadow edge
+    }
+    return a.concat(b);
+}
+
+function tToRad(date) {
+    date = (date !== undefined) ? new Date(date.getTime()) : new Date();
+    let milli = date - new Date(date.setHours(0, 0, 0, 0));
+    return (milli / DAY_MILLIS + 0.25) * TAU;
+}
+
+function draw() {
+    // work out how to display the current time
+    var d = new Date(),
+        a = tToRad(d);
+    var shape = moonShade(M_POS, SunCalc.getMoonIllumination(d));
+    var sTimes = SunCalc.getTimes(d, location.lat, location.lon);
+    var daylight = [sTimes.sunrise, sTimes.sunset];
+    //clear time area
+    g.setColor(0).fillRect(0, 176 - 45, 176, 176);
+    drawMoon(shape);
+    drawDayRing(daylight);
+    drawHHMM(d);
+    // draw pointer 
+    // TODO: Maybe later make this an overlay that can be removed?? -avoid drawing so much every minute/second
+    g.setColor(COL.needle).drawImage(needle, MX + RADII.needle * Math.cos(a), MY + RADII.needle * Math.sin(a), { rotate: a });
+
+}
+
+// Clear the screen once, at startup
+g.reset();
+// requires the myLocation app
+loadLocation();
+g.setBgColor(0, 0, 0).clear();
+// draw immediately at first
+draw();
+// now draw every second
+// eventually maybe update the moon just every hour??
+var secondInterval = setInterval(draw, 10000); //was 1000
+// Stop updates when LCD is off, restart when on
+Bangle.on('lcdPower', on => {
+    if (secondInterval) clearInterval(secondInterval);
+    secondInterval = undefined;
+    if (on) {
+        secondInterval = setInterval(draw, 10000); //was 1000
+        draw(); // draw immediately
+    }
+});
+/* Show launcher when middle button pressed
+This should be done *before* Bangle.loadWidgets so that
+widgets know if they're being loaded into a clock app or not */
+Bangle.setUI("clock");
+// Load widgets
+Bangle.loadWidgets();
+g.setTheme({ bg: "#000" });
```

Comment:
```suggestion
g.setTheme({fg:"#fff", bg:"#000", fg2:"#fff", bg2:"#004", fgH:"#fff", bgH:"#00f", dark:true });
```

So all theme stuff follow dark theme.

Reply:
Do I need to manually make this change and commit it on my fork? Is there any way to apply it/approve this suggestion right here?

Reply:
Yes - just click the `Commit suggestion` above on the suggestion and it will merge into the branch and PR :)

Reply:
I can do it here if you want. But maybe you'd like to see how it works for yourself?

Reply:
In the mobile app the button just says "Commit".

Reply:
Please do commit those changes. Thank you.

Reply:
Ok - good to know. Thanks!

Reply:
When I got back to a real computer, I did see a greyed out commit button (because it was already applied).Not sure why the mobile experience would differ between us. Thanks again!